File: ui/src/api/networkpolicy.js
import request from '@/utils/request'

export function listNetworkPolicies(cluster) {
  return request({
    url: `networkpolicy/${cluster}`,
    method: 'get',
  })
}

export function getNetworkPolicy(cluster, namespace, name, output='') {
  return request({
    url: `networkpolicy/${cluster}/${namespace}/${name}`,
    method: 'get',
    params: { output }
  })
}

export function deleteNetworkPolicies(cluster, data) {
  return request({
    url: `networkpolicy/${cluster}/delete`,
    method: 'post',
    data: data
  })
}

export function updateNetworkPolicy(cluster, namespace, name, yaml) {
  return request({
    url: `networkpolicy/${cluster}/update/${namespace}/${name}`,
    method: 'post',
    data: { yaml }
  })
}

export function updateNetworkPolicyObj(cluster, namespace, name, data) {
  return request({
    url: `networkpolicy/${cluster}/update_obj/${namespace}/${name}`,
    method: 'post',
    data: data
  })
}

File: pkg/server/views/kube_views/networkpolicy.go
package kube_views

import (
	"github.com/kubespace/kubespace/pkg/kube_resource"
	"github.com/kubespace/kubespace/pkg/server/views"
	"github.com/kubespace/kubespace/pkg/server/views/serializers"
	"github.com/kubespace/kubespace/pkg/utils"
	"github.com/kubespace/kubespace/pkg/utils/code"
	"net/http"
)

type NetworkPolicy struct {
	Views []*views.View
	*kube_resource.KubeResources
}

func NewNetworkPolicy(kr *kube_resource.KubeResources) *NetworkPolicy {
	d := &NetworkPolicy{
		KubeResources: kr,
	}
	vs := []*views.View{
		views.NewView(http.MethodGet, "/:cluster/:namespace/:name", d.get),
		views.NewView(http.MethodGet, "/:cluster", d.list),
		views.NewView(http.MethodPost, "/:cluster/delete", d.delete),
		views.NewView(http.MethodPost, "/:cluster/update/:namespace/:name", d.updateYaml),
	}
	d.Views = vs
	return d
}

func (d *NetworkPolicy) list(c *views.Context) *utils.Response {
	var ser serializers.ListSerializers
	if err := c.ShouldBindQuery(&ser); err != nil {
		return &utils.Response{Code: code.ParamsError, Msg: err.Error()}
	}
	reqParams := map[string]interface{}{
		"name":      ser.Name,
		"namespace": ser.Namespace,
	}
	return d.NetworkPolicy.List(c.Param("cluster"), reqParams)
}

func (d *NetworkPolicy) get(c *views.Context) *utils.Response {
	var ser serializers.GetSerializers
	if err := c.ShouldBindQuery(&ser); err != nil {
		return &utils.Response{Code: code.ParamsError, Msg: err.Error()}
	}
	reqParams := map[string]interface{}{
		"name":      c.Param("name"),
		"namespace": c.Param("namespace"),
		"output":    ser.Output,
	}
	return d.NetworkPolicy.Get(c.Param("cluster"), reqParams)
}

func (d *NetworkPolicy) delete(c *views.Context) *utils.Response {
	var ser serializers.DeleteSerializers
	if err := c.ShouldBind(&ser); err != nil {
		return &utils.Response{Code: code.ParamsError, Msg: err.Error()}
	}
	return d.NetworkPolicy.Delete(c.Param("cluster"), ser)
}

func (d *NetworkPolicy) updateYaml(c *views.Context) *utils.Response {
	var ser serializers.UpdateSerializers
	if err := c.ShouldBind(&ser); err != nil {
		return &utils.Response{Code: code.ParamsError, Msg: err.Error()}
	}
	reqParams := map[string]interface{}{
		"name":      c.Param("name"),
		"namespace": c.Param("namespace"),
		"yaml":      ser.Yaml,
	}
	return d.NetworkPolicy.UpdateYaml(c.Param("cluster"), reqParams)
}
